kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/AdaptiveBigQueryWriter.java

```diff
@@ -117,10 +134,15 @@ private boolean isTableMissingSchema(BigQueryException exception) {
         return writeResponse.getInsertErrors();
       }
       attemptCount++;
-      if (attemptCount >= AFTER_UPDATE_RETY_LIMIT) {
+      if (attemptCount >= AFTER_UPDATE_RETRY_LIMIT) {
         throw new BigQueryConnectException(
             "Failed to write rows after BQ schema update within "
-                + AFTER_UPDATE_RETY_LIMIT + " attempts for: " + tableId.getBaseTableId());
+                + AFTER_UPDATE_RETRY_LIMIT + " attempts for: " + tableId.getBaseTableId());
+      }
+      try {
+        Thread.sleep(RETRY_WAIT_TIME);
```

Comment:
If retry is caused by schema update, we don't need `sleep()`. I am think maybe we can refactor the code to 
```
try {
  bigQuery.insertAll(request);
} catch (BigQueryException exception) {
  if (isTableNotExisted) {
    attemptTableCreate(tableId, topic);
    retryInsert(request, maxRetry, INFINIT, 30)
  } else if (isTableMissingSchema) {
    attemptSchemaUpdate(tableId, topic);
    retryInsert(request, maxRetry, 5, 0)
  } else {
    throw exception;
  }
}

public retryInsert(request, maxRetry, waitTime)) {
  while (count < maxRetry) {
    try {
      bigQuery.insertAll(request);
      return;
    } catch (BigQueryException) {
      count++;
      sleep(waitTime);
      logger.info("I am sleeping")
    }
  }
}
```

Reply:
According to the comment here https://github.com/wepay/kafka-connect-bigquery/blob/master/kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/row/AdaptiveBigQueryWriter.java#L90-L91, updating schema in bigquery is also an async process, thus I think we need to wait for 30s for updating schema as well.

Reply:
nit: I would suggest add more comment in the function or clean up a little bit. It is not very easy to read to people unfamiliar.

Reply:
Cleaned up the existed comments to my best effort.